dev/bots/test.dart

```diff
@@ -245,224 +251,159 @@ Future<void> _runBuildTests() async {
-    workingDirectory: path.join(flutterRoot, relativePathToApplication),
-    expectNonZeroExit: expectNonZeroExit,
-    environment: <String, String>{
-      'FLUTTER_WEB': 'true',
-    },
-  );
-  print('Done.');
 }
 
 Future<void> _flutterBuildAot(String relativePathToApplication) async {
-  print('Running AOT build tests...');
+  print('${green}Testing AOT build$reset for $cyan$relativePathToApplication$reset...');
   await runCommand(flutter,
     <String>['build', 'aot', '-v'],
     workingDirectory: path.join(flutterRoot, relativePathToApplication),
-    expectNonZeroExit: false,
   );
-  print('Done.');
 }
 
 Future<void> _flutterBuildApk(String relativePathToApplication) async {
-  if (
-        (Platform.environment['ANDROID_HOME']?.isEmpty ?? true) &&
-        (Platform.environment['ANDROID_SDK_ROOT']?.isEmpty ?? true)) {
-    return;
-  }
-  print('Running APK build tests...');
+  print('${green}Testing APK --debug build$reset for $cyan$relativePathToApplication$reset...');
   await runCommand(flutter,
     <String>['build', 'apk', '--debug', '-v'],
     workingDirectory: path.join(flutterRoot, relativePathToApplication),
-    expectNonZeroExit: false,
   );
-  print('Done.');
 }
 
 Future<void> _flutterBuildIpa(String relativePathToApplication) async {
-  if (!Platform.isMacOS) {
-    return;
-  }
-  print('Running IPA build tests...');
+  assert(Platform.isMacOS);
+  print('${green}Testing IPA build$reset for $cyan$relativePathToApplication$reset...');
   // Install Cocoapods.  We don't have these checked in for the examples,
   // and build ios doesn't take care of it automatically.
   final File podfile = File(path.join(flutterRoot, relativePathToApplication, 'ios', 'Podfile'));
   if (podfile.existsSync()) {
     await runCommand('pod',
       <String>['install'],
       workingDirectory: podfile.parent.path,
-      expectNonZeroExit: false,
     );
   }
   await runCommand(flutter,
     <String>['build', 'ios', '--no-codesign', '--debug', '-v'],
     workingDirectory: path.join(flutterRoot, relativePathToApplication),
-    expectNonZeroExit: false,
   );
-  print('Done.');
 }
 
-Future<void> _runAdd2AppTest() async {
-  if (!Platform.isMacOS) {
-    return;
-  }
-  print('Running Add2App iOS integration tests...');
-  final String add2AppDir = path.join(flutterRoot, 'dev', 'integration_tests', 'ios_add2app');
-  await runCommand('./build_and_test.sh',
-    <String>[],
-    workingDirectory: add2AppDir,
-    expectNonZeroExit: false,
+Future<void> _flutterBuildDart2js(String relativePathToApplication, String target, { bool expectNonZeroExit = false }) async {
+  print('${green}Testing Dart2JS build$reset for $cyan$relativePathToApplication$reset...');
+  await runCommand(flutter,
+    <String>['build', 'web', '-v', '--target=$target'],
+    workingDirectory: path.join(flutterRoot, relativePathToApplication),
+    expectNonZeroExit: expectNonZeroExit,
+    environment: <String, String>{
+      'FLUTTER_WEB': 'true',
```

Comment:
Can you make sure that this cleanup will yield Cirrus tests that pass on dev, beta, and stable?  Right now, we have tests that are guaranteed to fail on those channels (e.g. because Flutter for web is disabled on non-master).

Reply:
I'm not sure how to test that short of actually doing a release.

Reply:
You could temporarily (with no intention of submitting) edit the tools logic for what channel you're on to synthesize fake runs of dev|beta|stable.  I leave it up to you on whether that churn of git push, wait, repeat is worth it.